table/block.cc

```diff
@@ -505,9 +532,81 @@ bool IndexBlockIter::PrefixSeek(const Slice& target, uint32_t* index) {
   }
 }
 
+// NOTE: in hash seek, if the key is not found in the restart intervals,
+// the iterator will simply be set as "invalid", rather than returning
+// the key that is just pass the target key.
+void DataBlockIter::HashSeek(const Slice& target) {
+  assert(data_block_hash_index_);
+  Slice user_key = ExtractUserKey(target);
+  uint8_t entry = data_block_hash_index_->Seek(user_key);
+
+  if (entry == kNoEntry) {
+    current_ = restarts_;  // not found, Invalidate the iterator
+    return;
+  }
+
+  if (entry == kCollision) { // HashSeek not effective
+    status_ = Status::NotSupported("Hash Collision");
```

Comment:
Status::NotSupported("Hash Collision") is a relatively expensive call. We should avoid this.
It is also a wrong status to use. "Not supported" is for a feature not implemented, not for normal fallback.
Why don't you return a boolean to indicate whether to fallback or not?